pkg/node/snapshot.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2025 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package node
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"slices"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/core/types"
+	archive "github.com/ethersphere/batch-archive"
+	"github.com/ethersphere/bee/v2/pkg/log"
+)
+
+type SnapshotLogFilterer struct {
```

Comment:
It would be good to add a line to check that SnapshotLogFilterer implements listener.BlockHeightContractFilterer interface, mainly for clarity.